.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,57 @@
+name: doxygen
+
+on:
+  push:
+    branches:
+      - master
+      - TODO
+      - release/*
+
+jobs:
+  Doxygen:
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+
+    steps:
+    - uses: actions/checkout@v2.0.0
+    - name: Install Depends
+      run: |
+        sudo apt install doxygen
+
+    - name: Generate Doxygen
+      shell: bash
+      run: |
+        mkdir build
+        cd build
+        cmake -DEVENT__DOXYGEN=ON ..
+        make doxygen
+
+    - name: Deploy Documentation
+      env:
+        LIBEVENT_DEPLOY_PRI: ${{ secrets.LIBEVENT_DEPLOY_PRI }}
```

Comment:
A was talking about this variable, is it available for PRs? (I couldn't find anything in documentation)

Reply:
https://github.com/libevent/libevent/pull/951#issue-365333549
‘secrets.LIBEVENT_DEPLOY_PRI’  is defined by myself.

Reply:
Indeed, the libevent repo will need the same, and my question is will these secrete variable will be available to the PRs?

For example, [in travis-ci](https://docs.travis-ci.com/user/environment-variables/#defining-encrypted-variables-in-travisyml) encrypted variables a not available to the PRs:

>Encrypted environment variables are not available to pull requests from forks due to the security risk of exposing such information to unknown code.

Reply:
`doxygen` workflow is not applicable for PRs. And the workflow runs only on push events, not `pull_request` events.
```
on:
  push:
    branches:
      - master
```
The workflow will be triggered if PR is merged.

I also test this in a [example](https://github.com/ygj6/github-actions-example/pull/1).

Reply:
>doxygen workflow is not applicable for PRs.

Ok, just out of curiosity do you know if user-defined secrets available for the PRs?

Reply:
I'm not sure. I will keep track of this issue, maybe I am missing something now.